puzzles/templatetags/puzzle_extras.py

```diff
@@ -114,11 +112,16 @@ def show_tags(puzzle, tag_form, request):
     context = {
         'puzzle': puzzle,
         'current_tags': current_tags,
-        'tag_form': tag_form,
+        'tag_form': TagForm(),
         'suggestions': suggestions
     }
     return context
 
+@register.inclusion_tag('show_tags.html')
+def show_tags(puzzle, request):
+    return create_tag_context(puzzle)
```

Comment:
I think it might be better to split create_tag_context up into the part that the templatetags needs for show_tags.html (current_tags and puzzle) and the part that tags_form.html needs(puzzle, suggestions, and tag_form). It feels weird to me that the views file imports templatetags

Reply:
Hmm, alright, I think you're right that we can factor this a little better. I'll create some tag helper methods in utils.py:
* get_tags(puzzle)
* get_all_tags()

and then have templatetags/puzzle_extra.spy and views.py import utils instead

Reply:
Refactored into a `tag_utils` module in latest revision.